addons/web/static/src/search/search_bar/search_bar.js

```diff
@@ -98,17 +99,39 @@ export class SearchBar extends Component {
             return;
         }
 
-        for (const searchItem of this.searchItems) {
+        // move properties under their related field
+        const searchItems = [...this.searchItems];
+        searchItems.sort((a, b) => (a.propertyItemId || a.id) - (b.propertyItemId || b.id));
+
+        for (const searchItem of searchItems) {
             const field = this.fields[searchItem.fieldName];
-            const type = field.type === "reference" ? "char" : field.type;
+            let type = field.type === "reference" ? "char" : field.type;
+
+            if (type === "properties" && searchItem.type === "field_property") {
+                // take the definition type instead
+                type = searchItem.definition.type
+
+                if (!this.state.expanded.includes(searchItem.propertyItemId)) {
+                    // properties field not unfolded, do not show the option
+                    continue;
+                }
```

Comment:
If I do that I'll need to duplicate the condition 
```
if (type === "properties" && searchItem.type === "field_property")
```

because I need to change the type with
```
type = searchItem.definition.type
```